ShortestPath/ShortestPath/Arc.cs

```diff
@@ -6,39 +6,5 @@ public class Arc
         public string Orig { get; set; }
         public string Dest { get; set; }
         public double Cost { get; set; }

```

Comment:
This class contains both the information about the topology and the "state" of the shortest path computation (the cost).
while the topology doesn't change, the cost is updated during the computation.
I would disentangle the two, for example you can have a dictionary, indexed by the node key holding the costs, this dictionary may be initialized (re created) each time you call the FindShortestPath method as it represent the state of the computation (and in some sense the result of the path search).

Reply:
Well, this is the Arc class and in the Arc class nothing will change even the Arc's Cost, but maybe this comment suits more to the Node class because,  there we have typology and the cost of the Nodes that is changing continuously.